clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -347,17 +347,21 @@ void executeCommandWithRetry(
       String topicName,
       int partition,
       IngestionCommandType command,
-      Supplier<Boolean> remoteCommandSupplier,
-      Runnable localCommandRunnable) {
+      Supplier<Boolean> isolatedProcessCommandSupplier,
+      Runnable mainProcessCommandRunnable) {
+    boolean isTopicPartitionHosted = isTopicPartitionHosted(topicName, partition);
+
+    // drop storage partition in main process if it does not exist
+    if (command == REMOVE_PARTITION && !isTopicPartitionHosted) {
+      mainProcessCommandRunnable.run();
+      return;
+    }
+
     do {
       if (isTopicPartitionHostedInMainProcess(topicName, partition)
-          || (!isTopicPartitionHosted(topicName, partition) && command != START_CONSUMPTION)) {
-        LOGGER.info(
-            "Executing command {} of topic: {}, partition: {} in main process process.",
-            command,
-            topicName,
-            partition);
-        localCommandRunnable.run();
+          || !isTopicPartitionHosted && command != START_CONSUMPTION) {
```

Comment:
nit: I'd suggest having a bracket here || ( condition && condition). Even though we know && has higher priority, this is somewhat not straightforward to people and might lead to some confusion.